### PR TITLE
bears/go: Add GoTypeBear

### PR DESCRIPTION
**Pull request by mr-karan**
This bear uses gotype to analyze Go code.

Fixes https://github.com/coala-analyzer/coala-bears/issues/275

### CODE REVIEW

File: bears/go/GoTypeBear.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import re
+
+from coalib.bearlib.abstractions.Lint import Lint
+from coalib.bears.LocalBear import LocalBear
+
+
+class GoTypeBear(LocalBear, Lint):
+    executable = 'gotype'
+    arguments = "-e {filename}"
```
Comment (Adrianzatreanu) [inline bears/go/GoTypeBear.py]: why is the argument ``-e`` needed?

Comment (mr-karan) [inline bears/go/GoTypeBear.py]: > -e
	report all errors (not just the first 10)
https://godoc.org/golang.org/x/tools/cmd/gotype

**Comment by Adrianzatreanu** [comment]
I honestly think that ``Add GoTypeBear`` as a shortlog would be enough, but I can't consider this a problem :)

File: tests/go/GoTypeBearTest.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from bears.go.GoTypeBear import GoTypeBear
+from tests.LocalBearTestHelper import verify_local_bear
+
+good_file = """package main
```
Comment (Adrianzatreanu) [inline tests/go/GoTypeBearTest.py]: as i've seen, as a codestyle, we don't use anything else on the same line as the triple backquotes. same applies to the bad_file as well :)

**Comment by Adrianzatreanu** [comment]
oh, you also need a rebase now :D

File: tests/go/GoTypeBearTest.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+from bears.go.GoTypeBear import GoTypeBear
+from tests.LocalBearTestHelper import verify_local_bear
```
Comment (Adrianzatreanu) [inline tests/go/GoTypeBearTest.py]: one more newline needed.

Comment (Adrianzatreanu) [inline tests/go/GoTypeBearTest.py]: we use 2 newlines between imports and the rest of the code.

Comment (mr-karan) [inline tests/go/GoTypeBearTest.py]: noted. :+1:

**Comment by Adrianzatreanu** [comment]
ack 92057d6

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Pull request merged**